File: Yarkool.RedisMQ/Dashboard/Monitoring/Response/MessageResponse.cs
namespace Yarkool.RedisMQ;

internal class MessageResponse
{
    /// <summary>
    /// MessageStatus
    /// </summary>
    public MessageStatus Status { get; set; }

    /// <summary>
    /// ExecutionTimes
    /// </summary>
    public int ExecutionTimes { get; set; }

    /// <summary>
    /// ErrorInfo
    /// </summary>
    public MessageErrorInfo? ErrorInfo { get; set; }

    /// <summary>
    /// Message
    /// </summary>
    public BaseMessage Message { get; set; } = default!;
}

File: Yarkool.RedisMQ/Dashboard/wwwroot/src/apis/response/MessageResponse.ts
import type { MessageStatus } from '@/types/MessageStatus'

export default interface MessageResponse {
  status: MessageStatus
  executionTimes: number
  message: MessageDataResponse
  errorInfo: MessageErrorInfoResponse
}

export interface MessageDataResponse {
  createTimestamp: number
  messageContent: string
  machineName: string
  delayTime: number
  messageId: string
  queueName: string
}

export interface MessageErrorInfoResponse {
  queueName: string
  groupName?: string
  consumerName: string
  exceptionMessage: string
  stackTrace?: string
  errorMessageContent?: string
  errorMessageTimestamp: number
}
